dev/src/document.ts

```diff
@@ -694,7 +694,12 @@ export class DocumentMask {
           }
         } else if (isPlainObject(value)) {
           extractFieldPaths(value, childPath);
-        } else {
+        } else if (value !== undefined) {
+          // If the value is undefined it can never participate in the document
+          // mask. With ignoreUndefinedProperties set to true, this shouldn't
+          // affect the document mask. With ignoreUndefinedProperties set to
+          // false, we'll throw an error when trying to create a proto from the
```

Comment:
Nit: This error is thrown first by `validateDocumentData`: "Cannot use "undefined" as a Firestore value (found in field "bar"). If you want to ignore undefined values, enable `ignoreUndefinedProperties`."

Reply:
Fixed.